src/SphereXEngine.sol

```diff
@@ -88,20 +106,27 @@ contract SphereXEngine is ISphereXEngine, AccessControlDefaultAdminRules {
      * @param rules bytes8 representing the new rules to activate.
      */
     function configureRules(bytes8 rules) external onlyOperator {
+        require(CF_AND_TXF_TOGETHER & rules != CF_AND_TXF_TOGETHER, "SphereX error: illegal rules combination");
         require(
-            RULES_1_AND_2_TOGETHER & uint64(rules) != RULES_1_AND_2_TOGETHER, "SphereX error: illegal rules combination"
+            CF_AND_SELECTIVE_TXF_TOGETHER & rules != CF_AND_SELECTIVE_TXF_TOGETHER,
+            "SphereX error: illegal rules combination"
         );
-        bytes8 oldRules = _engineRules;
-        _engineRules = rules;
-        emit ConfigureRules(oldRules, _engineRules);
+        require(
+            TXF_AND_SELECTIVE_TXF_TOGETHER & rules != TXF_AND_SELECTIVE_TXF_TOGETHER,
+            "SphereX error: illegal rules combination"
+        );
```

Comment:
bith checks in one go
```suggestion
        require(
            SELECTIVE_TXF & rules == SELECTIVE_TXF && rules != SELECTIVE_TXF,
            "SphereX error: illegal rules combination"
        );
```

Reply:
no because when i will add gas i will need to return this check as it is now